tests/integration/test_task_events_mgr.py

```diff
@@ -99,3 +106,53 @@ async def test__insert_task_job(flow, one_conf, scheduler, start, validate):
             i.submit_num for i
             in schd.data_store_mgr.added['jobs'].values()
         ] == [1, 2]
+
+
+async def test__always_insert_task_job(
+    flow, scheduler, mock_glbl_cfg, start, run
+):
+    """Insert Task Job _Always_ inserts a task into the data store.
+
+    Bug https://github.com/cylc/cylc-flow/issues/6172 was caused
+    by passing task state to data_store_mgr.insert_job: Where
+    a submission retry was in progress the task state would be
+    "waiting" which caused the data_store_mgr.insert_job
+    to return without adding the task to the data store.
+    """
+    global_config = """
+        [platforms]
+            [[broken1]]
+                hosts = no-such-host-1
+            [[broken2]]
+                hosts = no-such-host-2
+        [platform groups]
+            [[broken]]
+                platforms = broken1
+    """
+    mock_glbl_cfg('cylc.flow.platforms.glbl_cfg', global_config)
+
+    id_ = flow({
+        'scheduling': {'graph': {'R1': 'broken & broken2'}},
+        'runtime': {
+            'root': {'submission retry delays': 'PT10M'},
+            'broken': {'platform': 'broken'},
+            'broken2': {'platform': 'broken2'}
+        }
```

Comment:
Couldn't this be simplified to 1 task & platform?

Reply:
@wxtim could you add this as a comment?

Reply:
Still need to add code comment

Reply:
It's in the docstring.